pkg/controller/agent/controller_test.go

```diff
@@ -0,0 +1,151 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License 2.0;
+// you may not use this file except in compliance with the Elastic License 2.0.
+
+package agent
+
+import (
+	"context"
+	"reflect"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/tools/record"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	agentv1alpha1 "github.com/elastic/cloud-on-k8s/pkg/apis/agent/v1alpha1"
+	"github.com/elastic/cloud-on-k8s/pkg/controller/common"
+	"github.com/elastic/cloud-on-k8s/pkg/controller/common/comparison"
+	"github.com/elastic/cloud-on-k8s/pkg/utils/k8s"
+)
+
+func newReconcileAgent(objs ...runtime.Object) *ReconcileAgent {
+	r := &ReconcileAgent{
+		Client:   k8s.NewFakeClient(objs...),
+		recorder: record.NewFakeRecorder(100),
+	}
+	return r
+}
+
+func TestReconcileAgent_Reconcile(t *testing.T) {
+	type k8sFields struct {
```

Comment:
Nit: struct seems redundant and/or could be named differently to highlight this is the API Server state fixture.